pkg/sidecar/mock.go

```diff
@@ -0,0 +1,60 @@
+package sidecar
+
+import (
+	"context"
+	"errors"
+	"github.com/testground/sdk-go/sync"
+	gosync "sync"
+)
+
+// Runner
+func NewMockReactor() (Reactor, error) {
+	return &MockReactor{}, nil
+}
+
+// Reactor
+type MockReactor struct{}
+
+func (*MockReactor) Close() error { return nil }
+
+func (*MockReactor) Handle(ctx context.Context, handler InstanceHandler) error { return nil }
+
+func NewMockNetwork() *MockNetwork {
+	active := make([]string, 0)
+	configured := make([]*sync.NetworkConfig, 0)
+	mux := gosync.Mutex{}
+	return &MockNetwork{
+		Active:     active,
+		Configured: configured,
+		Closed:     false,
+		L:          &mux,
+	}
+
+}
+
+// Network
+type MockNetwork struct {
+	Active     []string
+	Configured []*sync.NetworkConfig
+	Closed     bool
+	L          gosync.Locker
```

Comment:
Let's use the mutex hat idiomatic pattern: https://dmitri.shuralyov.com/idiomatic-go#mutex-hat